src/plugins/intel_gpu/src/graph/include/lstm_seq_inst.h

```diff
@@ -16,6 +16,7 @@ struct typed_program_node<lstm_seq> : public typed_program_node_base<lstm_seq> {
 public:
     using parent::parent;
     ov::op::RecurrentSequenceDirection direction() const { return get_primitive()->direction; }
+    std::vector<size_t> get_shape_infer_dependencies() const override { return {1, 3}; }
```

Comment:
Looks like lstm_seq shape infer not use memory_deps, what is this change needed for?